execution/function/functions.go

```diff
@@ -120,6 +120,18 @@ var Funcs = map[string]FunctionCall{
 			F: math.Pi,
 		}
 	},
+	"timestamp": func(f FunctionArgs) promql.Sample {
+		if len(f.Samples) == 0 {
+			return InvalidSample
+		}
+		result := promql.Sample{}
+		for _, sample := range f.Samples {
+			if sample.H == nil { // only consider float samples
+				result.Points = append(result.Points, promql.Point{T: sample.T, V: float64(sample.T) / 1000})
```

Comment:
Hey, this cannot be implemented as a function i think, it will return the step timestamp for nested functions for example.